DirectProgramming/C++SYCL_FPGA/Tutorials/Features/experimental/mmhost/README.md

```diff
@@ -0,0 +1,335 @@
+```
+
+Find more information about how to navigate this part of the code samples in the [FPGA top-level README.md](/DirectProgramming/C++SYCL_FPGA/README.md).
+You can also find more information about [troubleshooting build errors](/DirectProgramming/C++SYCL_FPGA/README.md#troubleshooting), [running the sample on the Intel® DevCloud](/DirectProgramming/C++SYCL_FPGA/README.md#build-and-run-the-samples-on-intel-devcloud-optional), [using Visual Studio Code with the code samples](/DirectProgramming/C++SYCL_FPGA/README.md#use-visual-studio-code-vs-code-optional), [links to selected documentation](/DirectProgramming/C++SYCL_FPGA/README.md#documentation), etc.
+
+
+## Purpose
+When you design an IP component for an FPGA system, that system will often dictate the interface requirements of your IP component. This tutorial shows how to use `annotated_ptr` to configure Avalon memory-mapped host data interfaces. An Avalon memory-mapped host interface allows your IP component to send read or write requests to one or more Avalon memory-mapped agent interfaces. To learn more about Avalon memory-mapped host interfaces and Avalon memory-mapped agent interfaces, please refer to the appropriate section of the [Avalon Interface Specifications](https://www.intel.com/content/www/us/en/docs/programmable/683091/22-3/memory-mapped-interfaces.html). 
+
+![](assets/basic_avhost.svg)
+
+The compiler will infer Avalon memory-mapped host interfaces for your design when your kernel includes one or more pointer arguments. As with scalar kernel arguments, pointer arguments can be passed to your kernel via a `conduit` interface or your component's control/status register (CSR). By default, pointer arguments will be passed to your IP component through the CSR. For more details on kernel arguments, please see `<Temporary: https://github.com/intel-sandbox/wenkaixu.ip-auth-demos/tree/2024_0/mm-host>`. By default, the Intel® oneAPI DPC++/C++ Compiler will produce a kernel with a single Avalon memory-mapped host interface that will be shared amongst those pointers. 
+
+#### Example 1: A kernel with multiple pointer arguments
+```c++
+struct PointerIP{
+  int *x; 
+  int *y; 
+  int *z;
+  int size;
+
+  void operator()() const {
+    for (int i = 0; i < size; ++i) {
+      int mul = x[i] * y[i];
+      int add = mul + z[i];
+      y[i] = mul;
+      z[i] = add;
+    }
+  }
+};
+```
+![](assets/visio-naive.svg)
+
+### Using `annotated_ptr` to configure Avalon memory-mapped host interfaces
+
+You can override the default behaviour of a pointer argument by declaring an `annotated_ptr` kernel argument instead.
+
+#### Example 2: A kernel with a single customized Avalon memory-mapped host interface
+```c++
+struct SingleMMIP {
+  annotated_ptr<int, decltype(properties{
+    register_map,
+    buffer_location<kBL1>,
+    awidth<32>, 
+    dwidth<32>, 
+    latency<0>, 
+    read_write_mode<read>,
+    maxburst<4>, 
+    wait_request<true>
```

Comment:
We may have removed this now - talk to @tiwaria1

Reply:
Yup, simply remove the wait request property from all files.